### DISCUSSION TITLE
I would like to use this lib in C code, is there a simple way ?

### DISCUSSION THREAD

**Discussion by joyel24** [jgromes/RadioLib#1137] [Q&A]
I would like to use the Radiolib in this project witch is C code: 
https://github.com/nopnop2002/esp-idf-video-streaming

Do I need to edit all .h files to tells the compiler it's C++ code instead of C ?
Is there a simple way to use this lib for a C project ?

**Comment by jgromes** [comment]
At the moment, RadioLib does not support being directly used from C code, that would require a C API for all public C++ methods. It could be interesting to have that option, however, there are probably a few hundred public methods in RadioLib so it is a rather daunting task. However, using C from C++ code is much easier, so it would make more sense for you to use C++ as the main language and only use C for the parts that really need it.

> in this project witch is C code:

No it's not, it's a mix of C and C++, check the components folder.

**Comment by joyel24** [comment]
@jgromes Thanks for taking time to answer such an noob questiton ^^
I think I'm gonna make things different, For what I want to do, it'll be easier to take an esp cam and adding an rf module to it...
